src/Form/LegacyConsumer/FieldView.php

```diff
@@ -27,19 +27,38 @@ class FieldView {
 	 * @return void
 	 */
 	public static function render( Node $field, $formId ) {
-		$type = $field->getType();
-		$fieldIdAttribute = give( UniqueIdAttributeGenerator::class )->getId( $formId, $field->getName() );
+		$type                          = $field->getType();
+		$fieldIdAttribute              = give( UniqueIdAttributeGenerator::class )->getId( $formId, $field->getName() );
+		$visibilityConditionsAttribute = '';
+
+		if (
+			method_exists( $field, 'getVisibilityConditions' ) &&
+			( $conditions = $field->getVisibilityConditions() )
+		) {
+			$conditions                    = esc_attr( json_encode( $conditions ) );
+			$visibilityConditionsAttribute = "data-field-visibility-conditions=\"$conditions\"";
+		}
```

Comment:
This a minor stylistic suggestion, but I think something like this would be more clear, since I think re-assigning variables can be confusing.

```suggestion
		$visibilityConditions = method_exists( $field, 'getVisibilityConditions' ) ? $field->getVisibilityConditions() : null;

		if ( $visibilityConditions ) {
			$visibilityConditionsJson      = esc_attr( json_encode( $visibilityConditions ) );
			$visibilityConditionsAttribute = "data-field-visibility-conditions=\"$visibilityConditions\"";
		}
```